slither/visitors/slithir/expression_to_slithir.py

```diff
@@ -211,7 +212,31 @@ def _post_assignement_operation(self, expression):
                 # a = b = 1;
                 set_val(expression, left)
 
+    def _attempt_constant_folding(self, expression):
+        try:
+            const_fold = ConstantFolding(expression, expression.type)
+        except (NotConstant, AttributeError):
+            const_fold = None
+            return False
+        if const_fold is not None:
+            const_value = const_fold.result()
+            if expression.type in _signed_to_unsigned:
```

Comment:
Can you add some comments here why we are "guessing" the new_type. Slither's Expression AST doesn't propagate type information (the `expression.type` holds the kind of binary/unary operation, not the Solidity type). So, we don't have actual Solidity type here and need to guess the type for the Constant value.